test/uk/gov/hmrc/gform/gform/StructuredFormDataBuilderSpec.scala

```diff
@@ -240,13 +250,48 @@ class StructuredFormDataBuilderSpec extends Spec {
         "decField" -> "decFieldValue"
       ),
       objectStructure(
-        "field"    -> textNode("fieldValue"),
-        "ackField" -> textNode("ackFieldValue"),
-        "decField" -> textNode("decFieldValue")
+        field("field", textNode("fieldValue"), Map.empty),
+        field("ackField", textNode("ackFieldValue"), Map.empty),
+        field("decField", textNode("decFieldValue"), Map.empty)
       )
     )
   }
 
+  it must "provide the correct alternative field names for robotic XML" in {
+    def roboticsXmlPurposeMap(fieldName: String) = createStructuredPurposeMap(RoboticsXml, FieldName(fieldName))
+
+    validate(
+      createFormTemplate(
+        createNonRepeatingSection(
+          createAddress("field")
+        )
+      ),
+      createForm(
+        "field-street1"  -> "1",
+        "field-street2"  -> "2",
+        "field-street3"  -> "3",
+        "field-street4"  -> "4",
+        "field-uk"       -> "5",
+        "field-postcode" -> "6",
+        "field-country"  -> "7"
+      ),
+      objectStructure(
+        field(
+          "field",
+          objectStructure(
+            field("street1", textNode("1"), roboticsXmlPurposeMap("line1")),
+            field("street2", textNode("2"), roboticsXmlPurposeMap("line2")),
+            field("street3", textNode("3"), roboticsXmlPurposeMap("line3")),
+            field("street4", textNode("4"), roboticsXmlPurposeMap("line4")),
+            field("uk", textNode("5"), roboticsXmlPurposeMap("uk")),
+            field("postcode", textNode("6"), roboticsXmlPurposeMap("postcode")),
+            field("country", textNode("7"), roboticsXmlPurposeMap("country"))
+          ),
+          Map.empty
```

Comment:
Make our lives easy & remove the duplication by giving the alternateFieldNames parameter of field(...) a default, Map.empty.